Helm can use revision tags (#10112)
istio/istio.io: content/en/docs/setup/upgrade/helm/index.md

    $ helm upgrade istio-base manifests/charts/base -n istio-system --skip-crds
    {{< /text >}}

> 

Reviewer: I'm not sure we want to completely duplicate the content between here and the main canary upgrade page? Maybe we can describe the "concepts" on the canary page and then this page would just be how to use them with Helm. WDYT?

Author: That sounds MUCH better.

Author: Let's get a docs maintainer to agree to this before we do it. This will be in a later PR though.

Reviewer 2: Install/upgrade is one of the most commonly complained about parts of Istio, so yes, I'm very keen to see this improve in a later PR :)

> ### Revision Tags (experimental)

Reviewer 3: Let's use consistent heading across this and `istioctl` based install guide.

`Stable revision labels (experimental)`

Author: fixed

> {{< tip >}}

Reviewer 3: Warning instead of a tip?

Author: Warning is better. Fixed

> This is only supported with Helm charts from Istio 1.10+.

Reviewer 3: ```suggestion
This workflow is only supported from Istio versions 1.10 and above.
```

> {{< /warning >}}

Reviewer 2: This is only going into the 1.11 docs.  Are you implying "to upgrade from versions 1.10 and above" ?

Author: This is supported in Istio 1.10 as well. The hope was to get this merged into istio.io before we cut for 1.11, but that doesn't appear likely now. Users of 1.10 can do these steps as it's not a new feature.

Reviewer 4: I'm confused by this statement. It seems to imply that you must be upgrading `from` Istio 1.10 and above. You've used the boilerplates and show here an upgrade from 1.9 to 1.10. Maybe you mean `to` 1.10 and then from 1.9 or 1.8?

Author: I changed this to hopefully make it clear what helm supports going forward.

{{< boilerplate revision-tags-preamble >}}

{{< text bash >}}
$ helm template istiod manifests/charts/istio-control/istio-discovery -s templates/revision-tags.yaml --set revisionTags={prod-stable} --set revision=1-9-5 -n istio-system | kubectl apply -f -
$ helm template istiod manifests/charts/istio-control/istio-discovery -s templates/revision-tags.yaml --set revisionTags={prod-canary} --set revision=1-10-0 -n istio-system | kubectl apply -f -
{{< /text >}}

{{< warning >}}

> New `MutatingWebhookConfiguration` resources will be installed within your cluster, however, they are not owned by your Helm chart due to kubectl applying the templates. See the instructions

Reviewer 3: ```suggestion
These commands create new `MutatingWebhookConfiguration` resources in your cluster, however, they are not owned by any Helm chart due to `kubectl` manually applying the templates. See the instructions
```

    $ helm upgrade istio-base manifests/charts/base -n istio-system --skip-crds
    {{< /text >}}

### Stable revision labels (experimental)

{{< warning >}}

> Stable revision labels are only supported when updating Istio from and to Istio versions 1.10+.

Author: cc @ericvn This is supported in earlier versions with istioctl, but not with Helm. Hopefully this makes this clear.

$ helm template istiod manifests/charts/istio-control/istio-discovery -s templates/revision-tags.yaml --set revisionTags={prod-canary} --set revision=1-10-0 -n istio-system | kubectl apply -f -
{{< /text >}}

{{< warning >}}
These commands create new `MutatingWebhookConfiguration` resources in your cluster, however, they are not owned by any Helm chart due to `kubectl` manually applying the templates. See the instructions
below to uninstall revision tags.
{{< /warning >}}

{{< boilerplate revision-tags-middle >}}

{{< text bash >}}
$ helm template istiod manifests/charts/istio-control/istio-discovery -s templates/revision-tags.yaml --set revisionTags={prod-stable} --set revision=1-10-0 -n istio-system | kubectl apply -f -
{{< /text >}}

{{< boilerplate revision-tags-prologue >}}

#### Default tag

{{< boilerplate revision-tags-default-intro >}}

{{< text bash >}}
$ helm template istiod manifests/charts/istio-control/istio-discovery -s templates/revision-tags.yaml --set revisionTags={default} --set revision=1-10-0 -n istio-system | kubectl apply -f -
{{< /text >}}

> {{< boilerplate revision-tags-default-intro >}}

Reviewer 4: Probably another outro boilerplate.
